src/process/parent.rs

```diff
@@ -1,46 +1,72 @@
-use std::{io::Read, time::Duration};
+use std::io::Read;
 
+use super::{MAX_EVENTS, WAIT_FOR_CHILD};
+use crate::process::message::Message;
 use anyhow::{bail, Result};
 use mio::unix::pipe;
 use mio::unix::pipe::{Receiver, Sender};
 use mio::{Events, Interest, Poll, Token};
 
-use crate::process::message::Message;
-
+// Token is used to identify which socket generated an event
 const PARENT: Token = Token(0);
+
+/// Contains receiving end of pipe to child process and a poller for that.
 pub struct ParentProcess {
     receiver: Receiver,
     poll: Poll,
 }
 
+// Poll is used to register and listen for various events
+// by registering it with an event source such as receiving end of a pipe
 impl ParentProcess {
+    /// Create new Parent process structure
     pub fn new() -> Result<(Self, Sender)> {
+        // create a new pipe
         let (sender, mut receiver) = pipe::new()?;
+        // create a new poll, and register the receiving end of pipe to it
+        // This will poll for the read events, so when data is written to sending end of the pipe,
+        // the receiving end will be readable and poll wil notify
         let poll = Poll::new()?;
         poll.registry()
             .register(&mut receiver, PARENT, Interest::READABLE)?;
         Ok((Self { receiver, poll }, sender))
     }
 
+    /// Waits for associated child process to send ready message
+    /// and return the pid of init process which is forked by child process
     pub fn wait_for_child_ready(&mut self) -> Result<i32> {
-        let mut events = Events::with_capacity(128);
-        self.poll.poll(&mut events, Some(Duration::from_secs(5)))?;
+        // Create collection with capacity to store up to 128 events
```

Comment:
It is the same as this.
https://github.com/containers/youki/pull/64/files#r646003233